tsdb/engine/tsm1/reader.go

```diff
@@ -1323,6 +872,17 @@ func (m *mmapAccessor) readBlock(entry *IndexEntry, values []Value) ([]Value, er
 	return values, nil
 }
 
+func (m *mmapAccessor) readBytes(entry *IndexEntry, b []byte) ([]byte, error) {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+
+	if int64(len(m.b)) < entry.Offset+int64(entry.Size) {
+		return nil, ErrTSMClosed
+	}
+
+	return m.b[entry.Offset+4 : entry.Offset+int64(entry.Size)], nil
```

Comment:
I'd like to suggest a comment that this `+4` is about the CRC, just so it's clear.